pages/api/applications/[applicationId]/tasks/[taskId].ts

```diff
@@ -1,5 +1,163 @@
+  }
+  const taskId = taskOrError.id;
+
+  const { title, dueDate, notificationDateTime, isDone } = req.body;
+  const data: TaskPatchData = {
+    title,
+    dueDate,
+    isDone,
+    notificationDateTime: notificationDateTime ? new Date(notificationDateTime) : notificationDateTime,
+  };
+  const updatedTask: Task = await prisma.task.update({ where: { id: taskId }, data });
+  return res
+    .status(HttpStatus.OK)
+    .json(createJsonResponse(updatedTask, messages[MessageType.TASK_UPDATED_SUCCESSFULLY]));
 }
+
+async function handleDelete(uid: string, req: NextApiRequest, res: NextApiResponse<ApiResponse<EmptyPayload>>) {
+  const validationError = validatePathParameters(req);
+  if (validationError !== null) {
+    return res.status(HttpStatus.BAD_REQUEST).json(createJsonResponse({}, messages[validationError]));
+  }
+
+  const taskOrError = await getTask(uid, req);
+  if (taskOrError === MessageType.TASK_NOT_FOUND) {
+    return res.status(HttpStatus.NOT_FOUND).json(createJsonResponse({}, messages[taskOrError]));
+  }
+  const taskId = taskOrError.id;
+
+  await prisma.task.delete({ where: { id: taskId } });
+  return res.status(HttpStatus.OK).json(createJsonResponse({}, messages[MessageType.TASK_DELETED_SUCCESSFULLY]));
+}
+
+function validatePathParameters(req: NextApiRequest): Nullable<MessageType> {
+  const { taskId, applicationId } = req.query;
+  if (!taskId && !Number.isInteger(Number(taskId))) {
+    return MessageType.INVALID_TASK_ID;
+  }
+
+  if (!Number.isInteger(Number(applicationId))) {
+    return MessageType.INVALID_APPLICATION_ID;
+  }
+
+  return null;
+}
+
+function validatePatchRequest(req: NextApiRequest): Nullable<MessageType> {
+  const pathParameterError = validatePathParameters(req);
+  if (pathParameterError !== null) {
+    return pathParameterError;
+  }
+
+  const { title, dueDate, notificationDateTime, isDone } = req.body;
+
+  const isValidTitle = title === undefined || (typeof title === 'string' && !isEmpty(title));
+  if (!isValidTitle) {
+    return MessageType.INVALID_TITLE;
+  }
+
+  const isValidDueDate =
+    dueDate === undefined || isDate(dueDate) || (typeof dueDate === 'string' && isValidDate(dueDate));
+  if (!isValidDueDate) {
+    return MessageType.INVALID_DUE_DATE;
+  }
+
+  const isValidNotificationDateTime =
+    notificationDateTime === undefined ||
+    notificationDateTime === null ||
+    isDate(notificationDateTime) ||
+    (typeof notificationDateTime === 'string' && isValidDate('' + notificationDateTime));
+  if (!isValidNotificationDateTime) {
+    return MessageType.INVALID_NOTIFICATION_DATETIME;
+  }
+
+  const isValidDoneParameter = isDone === undefined || typeof isDone === 'boolean';
+  if (!isValidDoneParameter) {
+    return MessageType.INVALID_IS_DONE;
+  }
+
+  return null;
+}
+
+async function getTask(uid: string, req: NextApiRequest) {
+  const { applicationId, taskId } = req.query as unknown as TaskQueryParams;
```

Comment:
Do we need `as unknown`?
```suggestion
  const { applicationId, taskId } = req.query as TaskQueryParams;
```